### ISSUE TITLE
Not allowed to have a loginless system

### ISSUE THREAD

**Issue by pstn** [NixOS/nixpkgs#73003]
**Describe the bug**
There are reasons to want a system, that has no user that can log in by SSH or with their password, for example in a declarative container

**To Reproduce**
1. set up a minimal declarative container
2. set `users.mutableUsers = false;`
3. nix asserts that I will be locked out from the system and refuses to build the configuration with
```
building Nix...
building the system configuration...
error:
Failed assertions:
- Neither the root account nor any wheel user has a password or SSH authorized key.
You must set one to prevent being locked out of your system.
```

**Expected behavior**
nix generates me a container that I can log in with `nixos-container root-login $containername`, or maybe I don't want to log into it at all (because it's a disposable VM or something of the like). I think there should be a check whether the generated system is a container and an additional "I know what I'm doing" override for other cases. I'll create a PR once we agreed on what's best.

Maintainer information:
```
yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module: containers
```

Code causing the problem: https://github.com/NixOS/nixpkgs/blob/6445a7cf7d471e8629011b53d8a4a6999e68f7cc/nixos/modules/config/users-groups.nix#L576

mentioning @edolstra since you added the check.

**Comment by 7c6f434c** [comment]
>There are reasons to want a system, that has no user that can log in by SSH or with their password, for example in a declarative container

Is it an option to set a randomly-generated `hashedPassword` (the hash itself would be generated, not obtained by honestly hashing a password)? Recovering the password from such a hash is probably harder than a lot of other attack vectors.

But an override «doLockOutEveryone» sounds fine.

**Comment by pstn** [comment]
Setting a password would certainly be a possibility, but somewhat (for lack of a better word) really lame.
Having slept a night over it, I propose `users.disableSanityChecks`, that disables all the assertions at the bottom. I can see scenarios where two users with the same uid could be wanted. I want my unixoid to always give me enough rope to shoot myself in the leg.

**Comment by rnhmjoj** [comment]
What about not enabling ssh and disabling the getty?
It should completely lock anyone out, unless I am missing something. Take a look at the `nixos/modules/profiles/headless.nix` profile.

**Comment by pstn** [comment]
In case of a nixos declaratively defined nixos container, you are not logged out because the root system(d) can generate a root shell without any authentication, which is why I consider this a bug.

Having to "fool" the assertion to actually lock oneself out is another matter, I could live with that but I don't think it's nice behavior.

**Comment by AluisioASG** [comment]
This is still an issue.

**Comment by pstn** [comment]
Fixed via #73106
